onnxruntime/contrib_ops/cuda/transformers/generation_cuda_impl.cu

```diff
@@ -863,7 +863,7 @@ void KeyCacheExpansionKernelLauncher(const T* key_cache,
   equiv_head_size = (equiv_head_size & 1) == 0 ? (equiv_head_size >> 1) : equiv_head_size;
 
   // Here we know head_size is smaller than max_thread_num_per_block
-  int tpb = std::max(32, equiv_head_size);
+  int tpb = std::max(GPU_WARP_SIZE_HOST, equiv_head_size);
```

Comment:
why not use props.warpSize as well?

Reply:
props is not available here, it is not passed in as the previous function.

Reply:
Is it possible to pass props to be consistent?

Reply:
@tianleiwu, I don't want to do it at the moment. The problem is the interface design. The `KeyCacheExpansionKernelLauncher` is called in `ExpandBuffer`, which exists both in cpu and cuda

https://github.com/microsoft/onnxruntime/blob/9206b7cdc61df811ace631c25428fdf7f1a1b687/onnxruntime/contrib_ops/cpu/transformers/generation_device_helper.h#L355-L363

https://github.com/microsoft/onnxruntime/blob/9206b7cdc61df811ace631c25428fdf7f1a1b687/onnxruntime/contrib_ops/cuda/transformers/generation_device_helper.h#L142-L150

Adding `cudaDeviceProp` parameter will somehow pollute the cpu interface...